File: JavaOpenDataAccess/src/OpenDataAccess/Entity/IEntityAccess.java
package OpenDataAccess.Entity;

import OpenDataAccess.Data.IDataBase;
import OpenDataAccess.Data.IDataParameterList;
import OpenDataAccess.Data.IDataTransaction;
import OpenDataAccess.Data.IQuery;

import java.util.List;

public interface IEntityAccess {

    IDataBase GetDataBase();
    void SetDataBase(IDataBase value);
    EntityType GetEntityType();
    void SetEntityType(EntityType value);
    List<IEntityData> SelectEntities(IQuery query);
    List<IEntityData> SelectEntities(String procName, List<IDataParameterList> parameterList);
    List<IEntityData> SelectEntities(String entityName, String procName, List<IDataParameterList> parameterList);
    IEntityData SelectEntity(IQuery query);
    IEntityData SelectEntity(String procName, List<IDataParameterList> parameterList);
    IEntityData SelectEntity(String entityName, String procName, List<IDataParameterList> parameterList);
    IEntityData SelectEntityByPrimaryKey(Object primaryKey);
    Object InsertEntity(EntityType entityType, IEntityData entityData, IDataTransaction trans);
    Object InsertEntity(IEntityData entityData, IDataTransaction trans);
    Object InsertEntity(IEntity entity, List<String> fieldList, IDataTransaction trans);
    boolean UpdateEntityByPrimaryKey(Object primaryKey, IEntityData entityData, IDataTransaction trans);
    boolean UpdateEntityByPrimaryKey(Object primaryKey, IEntity entity, List<String> fieldList, IDataTransaction trans);
    boolean UpdateEntity(EntityType entityType, IQuery query, IEntityData entityData, IDataTransaction trans);
    boolean UpdateEntity(IQuery query, IEntityData entityData, IDataTransaction trans);
    boolean UpdateEntity(IQuery query, IEntity entity, List<String> fieldList, IDataTransaction trans);
    boolean DeleteEntityByPrimaryKey(Object primaryKey, IDataTransaction trans);
    boolean DeleteEntity(EntityType entityType, IQuery query, IDataTransaction trans);
    boolean DeleteEntity(IQuery query, IDataTransaction trans);
    IDataParameterList GetInParameter(Property property);
    IDataParameterList GetOutParameter(Property property);
    IDataParameterList GetInOutParameter(Property property);
}

File: Marriage/OpenDataAccessCore/OpenDataAccessCore.Entity/IEntityAccess.cs
using System.Collections.Generic;
using System.Data;
using OpenDataAccessCore.Data;

namespace OpenDataAccessCore.Entity
{
    public interface IEntityAccess
    {
        IDataBase CurrentDataBase { get; set; }
        EntityType EntityType { get; set; }
        List<IEntityData> SelectEntities(IQuery query);
        List<IEntityData> SelectEntities(string procName, List<IDbDataParameter> parameterList);
        List<IEntityData> SelectEntities(string entityName, string procName, List<IDbDataParameter> parameterList);
        IEntityData SelectEntity(IQuery query);
        IEntityData SelectEntity(string procName, List<IDbDataParameter> parameterList);
        IEntityData SelectEntity(string entityName, string procName, List<IDbDataParameter> parameterList);
        IEntityData SelectEntityByPrimaryKey(object primaryKey);
        bool InsertEntity(EntityType entityType, IEntityData entityData, out object primaryKey, IDbTransaction trans = null);
        bool InsertEntity(IEntityData entityData, out object primaryKey, IDbTransaction trans = null);
        bool InsertEntity(IEntity entity, List<string> fieldList, out object primaryKey, IDbTransaction trans = null);
        bool InsertEntity(EntityType entityType, IEntity entity, List<string> fieldList, out object primaryKey, IDbTransaction trans = null);
        bool UpdateEntityByPrimaryKey(object primaryKey, IEntityData entityData, IDbTransaction trans = null);
        bool UpdateEntityByPrimaryKey(object primaryKey, IEntity entity, List<string> fieldList, IDbTransaction trans = null);
        bool UpdateEntity(EntityType entityType, IQuery query, IEntityData entityData, IDbTransaction trans = null);
        bool UpdateEntity(IQuery query, IEntityData entityData, IDbTransaction trans = null);
        bool UpdateEntity(IQuery query, IEntity entity, List<string> fieldList, IDbTransaction trans = null);
        bool DeleteEntityByPrimaryKey(object primaryKey, IDbTransaction trans = null);
        bool DeleteEntity(EntityType entityType, IQuery query, IDbTransaction trans = null);
        bool DeleteEntity(IQuery query, IDbTransaction trans = null);
        IDbDataParameter GetInParameter(Property property);
        IDbDataParameter GetOutParameter(Property property);
        IDbDataParameter GetInOutParameter(Property property);

        IDbDataParameter InParameter(string name, object value);

        T SelectEntity<T>(IQuery query) where T : class;

        List<T> SelectEntities<T>(IQuery query) where T : class;
    }
}
